src/components/shopping.js

```diff
@@ -1,13 +1,35 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
 import ListItem from './ListItem';
+import Filter from './Filter';
 
 function Shopping(props) {
+  const [filterString, setFilterString] = useState('');
+  const [filteredList, setFilteredList] = useState(props.list);
+  useEffect(() => {
+    setFilteredList(props.list);
+  }, [props.list]);
+  const handleTextChange = event => {
+    setFilterString(event.target.value);
+    const newList = props.list.filter(item =>
+      item.name.toLowerCase().includes(event.target.value.toLowerCase()),
+    );
+    setFilteredList(newList);
+  };
+  const handleClear = () => {
+    setFilterString('');
+    props.list.length > 0 && setFilteredList(props.list);
```

Comment:
This is interesting! I'm not sure what it does . . . I'm going to have to learn this one.